batch-models/src/main/scala/org/ekstep/analytics/dashboard/CompetencyMetricsModel.scala

```diff
@@ -321,9 +322,131 @@ object CompetencyMetricsModel extends IBatchModelTemplate[String, DummyInput, Du
+          .join(totalLearningHoursTillYesterdayByOrgModified, totalLearningHoursTillTodayByOrgModified("userOrgID") === totalLearningHoursTillYesterdayByOrgModified("userOrgID"), "inner")
+          .withColumn("userOrgID:today", totalLearningHoursTillTodayByOrgModified("userOrgID"))
+          .withColumn("totalLearningHours", totalLearningHoursTillTodayByOrgModified("totalLearningHoursToday") - coalesce(totalLearningHoursTillYesterdayByOrgModified("totalLearningHoursYesterday"), lit(0)))
+          .select("userOrgID:today", "totalLearningHours")
+           totalLearningHoursTodayByOrg.show()
+        val totalLearningHoursYesterdayByOrg = totalLearningHoursTillYesterdayByOrgModified
+          .join(totalLearningHoursTillDayBeforeYesterdayByOrgModified, totalLearningHoursTillYesterdayByOrgModified("userOrgID") === totalLearningHoursTillDayBeforeYesterdayByOrgModified("userOrgID"), "inner")
+          .withColumn("userOrgID:yesterday", totalLearningHoursTillYesterdayByOrgModified("userOrgID"))
+          .withColumn("totalLearningHours", totalLearningHoursTillYesterdayByOrgModified("totalLearningHoursYesterday") - coalesce(totalLearningHoursTillDayBeforeYesterdayByOrgModified("totalLearningHoursDayBeforeYesterday"), lit(0)))
+          .select("userOrgID:yesterday", "totalLearningHours")
+        totalLearningHoursYesterdayByOrg.show()
+        val totalLearningHoursYesterday: String = totalLearningHoursYesterdayByOrg.select(sum("totalLearningHours").cast("string")).collect()(0)(0).toString
+        val totalLearningHoursToday: String = totalLearningHoursTodayByOrg.select(sum("totalLearningHours").cast("string")).collect()(0)(0).toString
+        print(totalLearningHoursYesterday + "\n")
+        print(totalLearningHoursToday + "\n")
+        val totalCertificationsTillToday: String = allCourseProgramCompletionWithDetailsDF.where(expr("courseStatus IN ('Live') AND userStatus=1 AND dbCompletionStatus = 2 AND issuedCertificateCount > 0")).count().toString
+        val totalCertificationsTillYesterday: String = redisGetKeyValue("lhp_certificationsTillToday")
+        val totalCertificationsTillDayBeforeYesterday: String = redisGetKeyValue("lhp_certificationsTillYesterday")
+        val totalCertificationsToday: String = ((totalCertificationsTillToday.toFloat) - (totalCertificationsTillYesterday.toFloat)).toString
+        val totalCertificationsYesterday: String = ((totalCertificationsTillYesterday.toInt) - (totalCertificationsTillDayBeforeYesterday.toInt)).toString
+        print(totalLearningHoursToday + "\n")
+        print(totalLearningHoursYesterday + "\n")
+        val certificationsOfTheWeek = allCourseProgramCompletionWithDetailsDF.where(expr("courseStatus IN ('Live') AND userStatus=1 AND " + s"courseCompletedTimestamp > '${startOf7thDayString}' " + s"AND courseCompletedTimestamp < '${endOfCurrentDayString}' " + "AND dbCompletionStatus = 2 AND issuedCertificateCount > 0"))
```

Comment:
the string inside `expr()` can be re-written more simply as `expr(s"courseStatus IN ('Live') AND userStatus=1 AND courseCompletedTimestamp > '${startOf7thDayString}' AND courseCompletedTimestamp < '${endOfCurrentDayString}' AND dbCompletionStatus = 2 AND issuedCertificateCount > 0")`